core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -400,7 +398,9 @@ private[kafka] class Processor(val id: Int,
 
   private val newConnections = new ConcurrentLinkedQueue[SocketChannel]()
   private val inflightResponses = mutable.Map[String, RequestChannel.Response]()
-  private val metricTags = Map("networkProcessor" -> id.toString).asJava
+  val metricTags = Map(
+    "protocol" -> securityProtocol.name, "listener" -> listenerName.value, "networkProcessor" -> id.toString
+    ).asJava
```

Comment:
We could omit the securityProtocol altogether. Although that's another deviation from the KIP, as you say the listener is the main differentiator. 
I started from the need to distinguish between metrics, but in our setup we have no listener labels, just protocols. That's why we initially came up with a 3-level tagging